src/openloco/graphics/gfx.cpp

```diff
@@ -861,15 +867,46 @@ namespace openloco::gfx
     }
 
     // 0x004C5C69
+    // input:
+    // regs.ax: left
+    // regs.bx: top
+    // regs.dx: right
+    // regs.bp: bottom
     void set_dirty_blocks(int32_t left, int32_t top, int32_t right, int32_t bottom)
     {
```

Comment:
Setting of dirty blocks should be part of the drawing engine. This is so that drawing engines that do not need to perform this (such as a headless environment can nop this). Can you move this function into `SoftwareDrawingEngine` and put a caller like in `draw_dirty_blocks` that gets the software drawing engine.